File: csharp/NedapRetailApiClient/ExampleApplication/WorkflowExample.cs
using Nedap.Retail.Api.Workflow.V2;
using System;
using System.Collections.Generic;
using System.IO;
using System.Net;

namespace Nedap.Retail.Api.Example
{
    internal class WorkflowExample
    {
        public static void RunExample(Client client)
        {
            string location = "http://location-testing";

            WorkflowEvent workflow = new WorkflowEvent();
            workflow.Type = "cycle_count_started";
            workflow.EventTime = DateTime.Now;
            workflow.Location = location;
            workflow.EpcCount = 10;
            workflow.MessageIds = new List<string>() { "1", "2", "3", "4", "5", "6", "7", "8", "9", "10" };

            try
            {
                // capture
                Console.WriteLine("------------- Capture workflow event");
                string cap = client.WorkflowV2.Capture(workflow);
                Console.WriteLine("captured workflow events");
            }
            catch (WebException e)
            {
                e.Print();
            }

            Console.ReadKey();
        }
    }
}

File: java/examples/erparticle/src/main/java/com/nedap/retail/services/examples/WorkflowExample.java
package com.nedap.retail.services.examples;

import java.util.Arrays;

import org.joda.time.DateTime;

import com.nedap.retail.messages.Client;
import com.nedap.retail.messages.workflow.WorkflowEvent;
import com.sun.jersey.api.client.UniformInterfaceException;

public class WorkflowExample {

    private static final String LOCATION = "Store:Sales floor";

    public static void runExample(final Client client) {
        System.out.println("*** Workflow API example ***");

        final WorkflowEvent workflow = makeWorkflowEvent();

        try {
            // capture
            System.out.println("Capturing workflow events...");
            client.captureWorkflow(workflow);
            System.out.println("--- Done ---");

        } catch (final UniformInterfaceException e) {
            System.err.println("Server responded with an error: " + e.getResponse().getEntity(String.class));
        }
    }

    private static WorkflowEvent makeWorkflowEvent() {
        final WorkflowEvent workflow = new WorkflowEvent();
        workflow.setType("cycle_count_started");
        workflow.setEventTime(DateTime.now());
        workflow.setLocation(LOCATION);
        workflow.setEpcCount((long) 10);
        workflow.setMessageIds(Arrays.asList("abc-123", "def-456", "ghi-789"));
        return workflow;
    }
}